File: 01_Leetcode_FAANG/Abinash_Biswal/Jewels_and_Stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        HashSet<Character> hs = new HashSet<>();
        for(int i = 0 ; i < jewels.length() ; i++){
            hs.add(jewels.charAt(i));
        }
        int counter = 0;
        for(int j = 0 ; j < stones.length() ; j++){
            if(hs.contains(stones.charAt(j))){
                counter++;
            }
        }
        return counter;
    }
}

File: 01_Leetcode_FAANG/Bikash Dalai/Jewels_and_stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        map<char,int> m;
        for(int i=0;i<jewels.size();i++)
        {
            m[jewels[i]]++;
        }
        int count=0;
        for(int i=0;i<stones.size();i++)
        {
            if(m[stones[i]]>0)
            {
                count++;
            }
        }
        
        return count;
    }
};
